socorro/unittest/external/es/base.py

```diff
@@ -100,25 +87,36 @@ def get_tuned_config(self, sources, extra_values=None):
 class ElasticsearchTestCase(TestCaseWithConfig):
     """Base class for Elastic Search related unit tests"""
 
-    def setup_method(self, method):
-        super().setup_method(method)
+    def setup_method(self):
+        super().setup_method()
         self.config = self.get_base_config()
         self.es_context = ConnectionContext(self.config)
+        self.crashstorage = ESCrashStorage(config=self.get_tuned_config(ESCrashStorage))
 
         self.index_client = self.es_context.indices_client()
-
-        with self.es_context() as conn:
-            self.connection = conn
-
-        self.es_context.create_index(self.es_context.get_index_template())
-
-    def teardown_method(self, method):
-        # Clear the test indices.
-        self.index_client.delete(self.es_context.get_index_template())
-        super().teardown_method(method)
+        self.conn = self.es_context.connection()
+
+        # Delete everything there first
+        for index_name in self.es_context.get_indices():
+            print(f"setup: delete test index: {index_name}")
+            self.es_context.delete_index(index_name)
+
+        to_create = [
+            self.es_context.get_index_for_date(utc_now()),
+            self.es_context.get_index_for_date(utc_now() - timedelta(days=7)),
+        ]
+        for index_name in to_create:
+            print(f"setup: creating index: {index_name}")
+            self.es_context.create_index(index_name)
+
+    def teardown_method(self):
+        for index_name in self.es_context.get_indices():
+            print(f"teardown: delete test index: {index_name}")
+            self.es_context.delete_index(index_name)
+        super().teardown_method()
```

Comment:
I changed the `elasticsearch_index` variable to `testsocorro_%W` and added an appropriate `elasticsearch_index_regex` so that all the tests use that and it makes cleanup much easier and more correct.

The next evolution of this code is probably to add an elasticsearch pytest fixture that does this instead of embedding it all mysteriously in the test case class.